File: android/app/src/main/java/com/buddhist/uposatha/UposathaApp.java
package com.buddhist.uposatha;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
import com.buddhist.uposatha.audio.NewPipeExtractorService;
import org.schabi.newpipe.extractor.NewPipe;
import org.schabi.newpipe.extractor.downloader.Downloader;
// import org.schabi.newpipe.extractor.downloader.RequieredDownloader; // Typo in NewPipe?

public class UposathaApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // Initialize NewPipeExtractor
        NewPipeExtractorService.getInstance().initialize(this);

        // Create Notification Channel for Audio
        createNotificationChannel();
    }

    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                "uposatha_audio",
                "Dhamma Audio Playback",
                NotificationManager.IMPORTANCE_LOW
            );
            channel.setDescription("Background audio playback foreground service");
            
            NotificationManager manager = getSystemService(NotificationManager.class);
            if (manager != null) {
                manager.createNotificationChannel(channel);
            }
        }
    }
}

File: android/app/src/main/kotlin/com/buddhist/uposatha/UposathaApp.kt
package com.buddhist.uposatha

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build

class UposathaApp : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        createNotificationChannel()
    }

    companion object {
        lateinit var instance: UposathaApp
            private set
    }

    private fun createNotificationChannel() {
        val channel = NotificationChannel(
            "uposatha_audio_channel",
            "Dhamma Audio Playback",
            NotificationManager.IMPORTANCE_LOW
        ).apply {
            description = "Background audio playback foreground service"
        }

        val manager = getSystemService(NotificationManager::class.java)
        manager?.createNotificationChannel(channel)
    }
}
